### PR TITLE
Eager loading TypeScript tests

### PR DESCRIPTION
**Pull request by willsoto**
#1431

### CODE REVIEW

File: tests/ts/query-builder-api/eager-loading-methods.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    children: {
+      pets: true,
+      movies: true
+    }
+  });
+
+  await Person.query()
+    .withGraphFetched('children(selectNameAndId).[pets(onlyDogs, orderByName), movies]')
+    .modifiers({
+      selectNameAndId(builder) {
```
Comment (willsoto) [inline tests/ts/query-builder-api/eager-loading-methods.ts]: I wonder if we can do better here with the `builder` typings. It is currently `any`.

Comment (koskimas) [reply tests/ts/query-builder-api/eager-loading-methods.ts]: Really? Yep, we should definitely be able to do better. I can take a look.

File: typings/objection/index.d.ts [hunk 2]
```diff
@@ -1099,6 +1099,7 @@ declare namespace Objection {
     mergeContext: ContextMethod<this>;
 
     modify: ModifyMethod<this>;
+    modifiers: ModifyMethod<this>;
```
Comment (koskimas) [inline typings/objection/index.d.ts]: This type is not correct. Modifiers takes an object full of modifiers. Something like

```js
interface ModifiersMethod<QB extends AnyQueryBuilder> {
  (modifiers: Modifiers): QB
}
```

Comment (willsoto) [reply typings/objection/index.d.ts]: Thank you! I was gonna ask about it. 

On a side note, I really like that API :+1:

File: typings/objection/index.d.ts [hunk 3]
```diff
@@ -774,6 +774,10 @@ declare namespace Objection {
     ): SingleQueryBuilder<QB>;
   }
 
+  interface GraphExpressionObjectMethod<QB extends AnyQueryBuilder> {
+    (): any;
```
Comment (willsoto) [inline typings/objection/index.d.ts]: I really think we can improve the types here. I was playing with `ModelRelations` but couldn't quite get it right. Any suggestions for this? I think we can access the current Model via `QB`?

Comment (koskimas) [reply typings/objection/index.d.ts]: We probably could, but I don't think it's worth the trouble at this point. I'd guess this method is rarely used.

Comment (willsoto) [reply typings/objection/index.d.ts]: Sounds good, I will leave as-is.

**Comment by willsoto** [comment]
@koskimas Are you able to retry the build please?

**Comment by koskimas** [comment]
Sure. I should really iron out those randomly failing tests.

**Comment by willsoto** [comment]
@koskimas Thanks! CI passed so this should be good for review/merge 😄

**Pull request merged**